File: 110. Balanced Binary Tree.py
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        global res = True

        def countHeight(root: TreeNode | None) -> bool:
            if not root or not res:
                return 0
            left = countHeight(root.left)
            right = countHeight(root.right)
            if abs(left - right) > 1:
                res = False
            return max(left, right) + 1
        countHeight(root)

        return res

File: 110. Balanced Binary Tree.java
class Solution {
    private boolean res = true;
    public boolean isBalanced(TreeNode root) {
        countHeight(root);
        return res;
    }
    public int countHeight(TreeNode root){
        if(root == null || !res) return 0;
        int left = countHeight(root.left);
        int right = countHeight(root.right);
        if(Math.abs(left - right) > 1) res = false;

        return Math.max(left, right) + 1;
    }
}